src/yunohost/settings.py

```diff
@@ -235,3 +243,44 @@ def _save_settings(settings, location=SETTINGS_PATH):
             settings_fd.write(result)
     except Exception as e:
         raise YunohostError('global_settings_cant_write_settings', reason=e)
+
+
+# Meant to be a dict of setting_name -> function to call
+post_change_hooks = {}
+
+
+def post_change_hook(setting_name):
+    def decorator(func):
+        assert setting_name in DEFAULTS.keys(), "The setting %s does not exists" % setting_name
+        assert setting_name not in post_change_hooks, "You can only register one post change hook per setting (in particular for %s)" % setting_name
```

Comment:
I understand this cover use case for #640. But in other use case it could be useful to have a list of hook functions associated with a setting name. What do you think ?

Reply:
Hmyea I was a bit lazy to think about that so I just added this constrain :sweat_smile: 

My idea is that it should be easy to implement once we need it

Reply:
I see you have already done it ! 😄  I cannot fix the documentation at the top of this PR but may be it's not very important.

Reply:
Uuuuh, not sure we're talking about the same thing ? 

Ain't we talking about the ability or not to have multiple @post_change_hook("name") for the same "name" ?

e.g. 

```python
@post_change_hook("foo")
def foo1(...):
   ...

@post_change_hook("foo")
def foo3(...):
   ...
```

?

Reply:
Sorry I messed up. I was indeed talking first about having multiple hook functions for the same setting but it’s not mandatory.
Then I switched to talking about having the setting name as a parameter.